doctest/test_docs.py

```diff
@@ -0,0 +1,179 @@
+
+from functools import partial
+from odfe_sql_cli.esconnection import ESConnection
+from odfe_sql_cli.utils import OutputSettings
+from odfe_sql_cli.formatter import Formatter
+from elasticsearch import Elasticsearch, helpers
+
+ENDPOINT = "http://localhost:9200"
+ACCOUNTS = "accounts"
+EMPLOYEES = "employees"
+
+
+class DocTestConnection(ESConnection):
+
+    def __init__(self, query_language="sql"):
+        super(DocTestConnection, self).__init__(endpoint=ENDPOINT, query_language=query_language)
+        self.set_connection()
+
+        settings = OutputSettings(table_format="psql", is_vertical=False)
+        self.formatter = Formatter(settings)
+
+    def process(self, statement):
+        data = self.execute_query(statement, use_console=False)
+        output = self.formatter.format_output(data)
+        output = "\n".join(output)
+
+        click.echo(output)
+
+
+def pretty_print(s):
+    try:
+        d = json.loads(s)
+        print(json.dumps(d, indent=2))
+    except json.decoder.JSONDecodeError:
+        print(s)
+
+
+sql_cmd = DocTestConnection(query_language="sql")
+ppl_cmd = DocTestConnection(query_language="ppl")
+test_data_client = Elasticsearch([ENDPOINT], verify_certs=True)
+
+
+def sql_cli_transform(s):
+    return u'sql_cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+
+
+def ppl_cli_transform(s):
+    return u'ppl_cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+
+
+def bash_transform(s):
+    # TODO: add ppl support, be default cli uses sql
+    if s.startswith("odfesql"):
+        s = re.search(r"odfesql\s+-q\s+\"(.*?)\"", s).group(1)
+        return u'cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+    return (r'pretty_print(sh("""%s""").stdout.decode("utf-8"))' % s) + '\n'
+
+
+sql_cli_parser = zc.customdoctests.DocTestParser(
+    ps1='od>', comment_prefix='#', transform=sql_cli_transform)
+
+ppl_cli_parser = zc.customdoctests.DocTestParser(
+    ps1='od>', comment_prefix='#', transform=ppl_cli_transform)
+
+bash_parser = zc.customdoctests.DocTestParser(
+    ps1=r'sh\$', comment_prefix='#', transform=bash_transform)
+
+
+def set_up_accounts(test):
+    set_up(test)
+    load_file("accounts.json", index_name=ACCOUNTS)
+
+
+def load_file(filename, index_name):
+    filepath = "./test_data/" + filename
+
+    # generate iterable data
+    def load_json():
+        with open(filepath, "r") as f:
+            for line in f:
+                yield json.loads(line)
+
+    # Need to enable refresh, because the load won't be visible to search immediately
+    # https://stackoverflow.com/questions/57840161/elasticsearch-python-bulk-helper-api-with-refresh
+    helpers.bulk(test_data_client, load_json(), stats_only=True, index=index_name, refresh='wait_for')
+
+
+def set_up(test):
+    test.globs['sql_cmd'] = sql_cmd
+    test.globs['ppl_cmd'] = ppl_cmd
+
+
+def tear_down(test):
+    # drop leftover tables after each test
+    # TODO: delete all will potentially also delete AES FGAC metadata index
```

Comment:
I think we didn't delete _all now, right?

Reply:
Right, here we only delete the test indexes. Will remove the todo here